src/objects/impls/ruin.rs

```diff
@@ -0,0 +1,11 @@
+use crate::{
+    macros::*,
+    objects::{Ruin, Structure},
+};
+
+simple_accessors! {
+    impl Ruin {
+        pub fn structure() -> Structure = structure;
```

Comment:
In further testing, calling this panics - I think due to needing to convert with `__structure_type_str_to_num`, but I could use an assist in how to hook this in correctly.

Reply:
The way `Structure` is built, it _shouldn't_ panic when used this way. @daboross any idea?
The `simple_accessor` calls `js_unwrap` which in turns use the `TryFrom` definition. My only guess would be that `TryFrom` returns `ConversionError`, 
In turn, it might be because those structure are "vanilla", ie without a structure type in the list we have?

@shanemadden do you have a panic trace?

Reply:
Seems like the `structure` method unwraps an `Err`. I'm really not sure why though.